tests/unit/python/foglamp/common/test_configuration_manager.py

```diff
@@ -0,0 +1,430 @@
+import asyncio
+import pytest
+from foglamp.common.configuration_manager import ConfigurationManager
+from foglamp.common.configuration_manager import ConfigurationManagerSingleton
+from foglamp.common.storage_client.storage_client import StorageClient
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+class TestConfigurationManager():
+    def test_constructor_no_storage_client_defined_no_storage_client_passed(self):
+        with pytest.raises(TypeError) as excinfo:
+            ConfigurationManager()
+        assert 'Must be a valid Storage object' in str(excinfo.value)
+
+    def test_constructor_no_storage_client_defined_storage_client_passed(self):
+        storageClientMock = MagicMock(spec=StorageClient)
+        c = ConfigurationManager(storageClientMock)
+        assert hasattr(c, '_storage')
+        assert c._storage is storageClientMock
+        assert hasattr(c, '_registered_interests')
+        ConfigurationManagerSingleton._shared_state = {}
```

Comment:
let's move to method  level setup and teardown? that way even if any test gets error / failed, it will keep state for other test clean